cmake/scripts/common/Macros.cmake

```diff
@@ -708,6 +708,10 @@ macro(core_find_versions)
     string(TOLOWER ${APP_VERSION_TAG} APP_VERSION_TAG_LC)
   endif()
   string(REPLACE "." "," FILE_VERSION ${APP_ADDON_API}.0)
+  # Let Debian builders etc override APP_ADDON_REPOS
+  if(OVERRIDE_ADDON_REPOS)
+    set(APP_ADDON_REPOS ${OVERRIDE_ADDON_REPOS})
+  endif()
```

Comment:
You already patch kodi extensively from what i understand, so why not just patch in your "repo" into https://github.com/xbmc/xbmc/blob/178dc39f30f992291e55a97c582f4cfe2980a248/version.txt#L14

Reply:
On this note, does a file:// URI work as is i wonder?
No idea what value your attempting to use to designate the "local" repo.

Reply:
Patching version.txt is undesirable as it changes more frequently than CMake code.

file:// works with CCurlFile but not with CFile. To make file:// work, URIUtils::Substitution or sonething must be patched, too

Reply:
The value for local repo is described in the issue I linked. If Kodi has mirrors.kodi.tv disabled or purely offline / intranet, how should language addons be available? Previous attempt was to install them all but users started complaining about unused language addons keep updating in `$HOME/.kodi/addons` Local repo solves this question quite elegantly.

Reply:
I would highly suggest you provide relevant info in the PR description at least. All i see in your link is a half dozen other links.

What is the format you will use to indicate a "local" repo?

Reply:
> Patching version.txt is undesirable as it changes more frequently than CMake code.
> 
> file:// works with CCurlFile but not with CFile. To make file:// work, URIUtils::Substitution or sonething must be patched, too

So does file:// mean you dont need to do any of the other changes in this PR? The original code is a CCurlFile after all?

Reply:
Well, I tested with file:// protocol, had to debug stuff till I found that `file:///usr/share/kodi/language-repo` gets substituted into `file://usr/…` (notice eaten slash between file and usr!), the uri becomes invalid and addon retrieval fails. I switched to special protocol with this change and vooila, it finally worked.

Reply:
Please note that in order for file:// protocol to start workimg, I also had to patch FetchChecksum into CCurlFile. So:

for special protocol to work, both CRepository::Fetch{Checksum,Index} must be CFile,
for file:// protocol, both CRepository::Fetch{Checksum,Index} must be CCurlFile AND `URIUtils::PerformSubstitution` should be patched notvto transform file protocol links

Reply:
Updated PR description with relevant info